internal/code-generator/generator/generator_test.go

```diff
@@ -283,6 +284,23 @@ func (s *GeneratorTestSuite) TestGenerationFiles() {
 		assert.NoError(s.T(), err)
 		out, err := ioutil.ReadFile(outPath.Name())
 		assert.NoError(s.T(), err)
-		assert.EqualValues(s.T(), gold, out, fmt.Sprintf("%v: %v, expected: %v", test.name, string(out), string(gold)))
+
+		expected := withoutSpaces(string(gold))
+		actual := withoutSpaces(string(out))
+		assert.Equal(s.T(), expected, actual, "Test %s", test.name)
+	}
+}
+
+// withoutSpaces is a helper to compare content irregarding of whitespaces.
```

Comment:
I think this could more easily be expressed as

```go
return strings.Join(strings.Fields(str), " ")
```

Reply:
True, but the performance is different:

```
$ go test -bench='BenchmarkWhitespace*' .
goos: darwin
goarch: amd64
pkg: gopkg.in/src-d/enry.v1
BenchmarkWhitespaceJoin-4      	 5000000	       333 ns/op	     192 B/op	       3 allocs/op
BenchmarkWhitespaceBuilder-4   	 5000000	       366 ns/op	      64 B/op	       1 allocs/op
PASS
```

although it's obviously is not most perf critical part, but I would prefer keeping it this way if possible.

Reply:
I'd argue that a difference of 33ns/op on a test helper isn't worthwhile, in contrast to more easily-understood code. But I am not deeply attached to this—I think the more important aspect is to not drop the spaces entirely, because if you do the comparisons could be made incorrect (even though in this case it's unlikely to occur because of the inputs we are using).

FWIW I wrote a token-aware differencing tool for Go source that we could also use, because we ran into this issue with Kythe too—but that might be overkill.